src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractWalletController.cs

```diff
@@ -135,13 +135,15 @@ public IActionResult GetHistory(string walletName, string address)
                 {
                     TransactionData transaction = item.Transaction;
 
+                    bool isCoinBase = transaction.IsCoinBase.HasValue && transaction.IsCoinBase.Value;
+
                     // Record a receive transaction
                     transactionItems.Add(new ContractTransactionItem
                     {
                         Amount = transaction.Amount.ToUnit(MoneyUnit.Satoshi),
                         BlockHeight = transaction.BlockHeight,
                         Hash = transaction.Id,
-                        Type = ContractTransactionItemType.Received,
+                        Type = isCoinBase ? ContractTransactionItemType.GasRefund : ContractTransactionItemType.Received,
```

Comment:
For miner wallets this will say GasRefund when it's the MinerReward. Just checked the Payment and TransactionData classes, don't know that there's anything we can do about it - to change it we would need to TxOut index

Reply:
We can use `Index` on the `TransactionData` object to get whether this is a miner reward (index == 0)